ref(event_manager): Fix typing issues for event_manager

armenzg opened the pull request:
We want to make a lot of changes to event_manager and we need to have backend typing in place for the upcoming work.

Fixes #52877

armenzg committed:
feat(typing): Type event_manager

priscilawebdev committed:
fix(getting-started-docs): Fix path native qt (#52878)

loewenheim committed:
fix: Return more than one bundle containing a URL (#52868)

Consider the following situation:
1. Symbolicator needs the bundle for release `foo`, dist `1` that
contains `https://example.com/main.dart.js`.
2. URL search is not exact. Symbolicator asks the `artifact-lookup`
endpoint for a bundle with release `foo`, dist `1`, URL `/main`.
3. There are two bundles for the release/dist. Bundle 1 contains
`main.dart`, bundle 2 contains `main.dart.js`. It so happens that bundle
1 is newer.
4. Bundle 1 gets returned to Symbolicator.
5. The bundle Symbolicator got back doesn't contain the file it wanted.

We fix this by returning not just the first bundle that contains the
URL, but up to `MAX_BUNDLES_QUERY` (=5) bundles.

asottile-sentry committed:
ref: use django 3.x json field directly (#52812)

resolves a migration warning: `fields.W904`

```
sentry.ReleaseActivity.data: (fields.W904) django.contrib.postgres.fields.JSONField is deprecated. Support for it (except in historical migrations) will be removed in Django 4.0.
	HINT: Use django.db.models.JSONField instead.
```

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

k-fish committed:
fix(mobile): Elide invalid_data errors from Cocoa in the UI (#50952)

### Summary
Don't show the invalid_data errors in the UI as to not cause false alarm
as this has since been fixed.


### Screenshots

#### Before
![Screenshot 2023-06-14 at 12 08 45
PM](https://github.com/getsentry/sentry/assets/6111995/ccbef4f6-023e-4e1c-80cb-921f7d68ea55)

#### After
![Screenshot 2023-06-14 at 12 04 51
PM](https://github.com/getsentry/sentry/assets/6111995/24582d7a-058e-4b00-9740-9c88f1f513f6)

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

asottile-sentry committed:
ref: remove django_compat (#52879)

we are now on django 3+




<!-- Describe your PR here. -->

barkbarkimashark committed:
fix(helpful-event): fix querying for title for helpful event (#52838)

We were improperly mapping the dataset resulting in all calls with
`query` string filtering on the `IssuePlatform` dataset.

Resolves SENTRY-1379

armenzg committed:
ref(event_manager): Fix typing issues for event_manager

We want to make a lot of changes to event_manager and we need to have backend typing in place for the upcoming work.

Fixes #52877

spalmurray committed:
ref(stats): Convert TeamResolutionTime to functional (#52845)

- Remove unnecessary prop
- Convert teamResolutionTime to functional

leedongwei committed:
chore(roles): Default team-roles flag to True (#52830)

We should let self-hosted Sentry use [team-level
roles](https://github.com/getsentry/sentry/discussions/49028).

dashed committed:
chore(hybrid-cloud): Stabilize slack command test (#52767)

cmanallen committed:
feat(replays): Add multiprocessing consumer step (#48276)

To run the consumer issue the following command.
`ingest-replay-recordings --processes NUM_CORES * 4 --max-batch-size
100`. Additional options can be specified but currently have sane
defaults.

cathteng committed:
fix(hybrid-cloud): auth attach identity unsets idp:provisioned flag (#52846)

ryan953 committed:
fix(replay): Fix an issue where useInitialTimeOffsetMs doesn't consume numeric timestamps correctly (#52828)

Commit:
ref(ui): Change wording of stacktrace order in user settings (#52889)

Closes #50697

Update sort order wording in user settings to match issue details

Before: 
<img width="555" alt="Screenshot 2023-07-14 at 9 46 50 AM"
src="https://github.com/getsentry/sentry/assets/22582037/d61e854a-91d9-4641-bfed-d97f38ef4647">
After:
<img width="555" alt="Screenshot 2023-07-14 at 9 45 25 AM"
src="https://github.com/getsentry/sentry/assets/22582037/8aba23a8-049e-4bac-bd31-9678895cd04c">

ceorourke committed:
ref(sort): Clean up priority/better priority (#52769)

Remove checks for the feature flag and any code written specifically to
use the feature flag, as well as remove references to the old priority
sort. We do have 310 saved searches using the old priority so if they
use it it will show "None" in the sort dropdown temporarily, but I
checked in Redash and nobody's default search is set to it.

Backend PR to follow: https://github.com/getsentry/sentry/pull/52770

armenzg committed:
More fixes

armenzg commented on src/sentry/event_manager.py:
```diff
@@ -2155,7 +2157,7 @@ def save_attachment(
         headers={"Content-Type": attachment.content_type},
     )
-    file.putfile(BytesIO(data), blob_size=settings.SENTRY_ATTACHMENT_BLOB_SIZE)
+    file.putfile(BytesIO(data), blob_size=settings.SENTRY_ATTACHMENT_BLOB_SIZE)  # type: ignore
```
I was getting this `src/sentry/event_manager.py:2158: error: Argument 1 to "BytesIO" has incompatible type "Callable[[CachedAttachment], Any]"; expected "Buffer"  [arg-type]`

I could not figure it out.

armenzg commented on src/sentry/event_manager.py:
```diff
@@ -2266,7 +2268,8 @@ def _calculate_event_grouping(
 
-    hashes.write_to_event(event.data)
+    # Using cast to satisfy mypy
+    hashes.write_to_event(cast(Dict[str, Any], event.data))
```
I was not sure how to fix this one without getting complicated.
I was getting this message before: `src/sentry/event_manager.py:2269: error: Argument 1 to "write_to_event" of "CalculatedHashes" has incompatible type "NodeData"; expected "Dict[str, Any]"  [arg-type]`

Here's the code for [write_to_event](https://github.com/getsentry/sentry/blob/db9df6f3a513c9ebc821d0382e2b7e78637696bb/src/sentry/grouping/result.py#L100)

asottile-sentry replied:
`cast` should only be used for mypy bugs -- you'll need to fix the types here or convert -- probably by fixing the signature for `write_to_event`

getsantry[bot] committed:
:knife: regenerate mypy module blocklist

barkbarkimashark committed:
fix(group_attributes): import attributes when django starts (#52844)

We weren't logging anything to ddog since the module was never imported,
this should fix that

ceorourke committed:
ref(alerts): Remove mute metric alerts flag (#52823)

Remove the feature flag and usages of it since the feature is in GA now.
This must be merged after the [front end
PR](https://github.com/getsentry/sentry/pull/52822) and the [getsentry
PR](https://github.com/getsentry/getsentry/pull/11117).

michellewzhang committed:
fix(replays) make title and issue links clickable (#52894)

Fixes getsentry/team-replay#113 by making the Title and Issue links in
replay clickable and linking to the event ID issue page.

Before:


https://github.com/getsentry/sentry/assets/187460/a39245b8-f13d-4acb-8dec-a92b8e2214a9



After:


https://github.com/getsentry/sentry/assets/187460/c6baf600-41e0-43ec-bbcc-6de163e49c4a

gggritso committed:
ref(starfish): Simplify `EventView` creation (#52376)

It's redundant to pass both a `location` and `pageFilters` object when
creating event views for Starfish. The reason we pass both is that
there's no way to auto-determine an interval from a `location`, and no
way to create an event view from a `pageFilters`. So, I added a way to
create an `EventView` from a `pageFilters`, and now we don't need to
pass `location` around anymore in places where we both have to create
event views and determine the intervals.

Also, `location` is a primitive, IMO it's better to use `pageFilters`,
it contains the stuff we want. This is a bit of cleanup post removing
hard-coded project IDs.

Co-authored-by: Dominik Buszowiecki <44422760+DominikB2014@users.noreply.github.com>

asottile-sentry commented on src/sentry/eventtypes/__init__.py:
```diff
@@ -1,3 +1,13 @@
+__all__ = [
+    "CspEvent",
+    "DefaultEvent",
+    "ErrorEvent",
+    "ExpectCTEvent",
+    "ExpectStapleEvent",
+    "HpkpEvent",
+    "TransactionEvent",
+]
```
I would remove this personally -- it's better to import explicitly since `__init__.py` makes it much easier to introduce import cycles and to worsen startup performance

asottile-sentry committed:
ref: update missed django monkeypatches for django 3.x (#52886)

<!-- Describe your PR here. -->

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

cmanallen committed:
fix(replays): Comment out desynced tests (#52900)

scttcper committed:
feat(grouping): Remove issue details grouping tab 🪦 (#52895)

ceorourke committed:
ref(sort): Remove priority and betterpriority flags (#52770)

Remove feature flag / references for better priority sort as it's in GA.
This also routes the "priority" sort to use "betterPriority" temporarily
while I clean up references on the front end.

Front end PR must be merged first:
https://github.com/getsentry/sentry/pull/52769

mifu67 committed:
feat(opsgenie): Add basic installation flow (#52577)

Add a basic installation flow for the Opsgenie integration (see provided
screenshots).

Fixes ER-1696

<img width="1512" alt="Screenshot 2023-07-11 at 3 52 36 PM"
src="https://github.com/getsentry/sentry/assets/83109586/b9ee211e-fe66-411e-ab02-8f373e9e92ca">
<img width="1512" alt="Screenshot 2023-07-11 at 3 52 49 PM"
src="https://github.com/getsentry/sentry/assets/83109586/41a1555b-45f1-46ad-9796-dfd365dcbeb7">
<img width="1512" alt="Screenshot 2023-07-11 at 3 53 04 PM"
src="https://github.com/getsentry/sentry/assets/83109586/5710de1c-1ca5-4d96-8b96-806ccd7c7954">
<img width="1512" alt="Screenshot 2023-07-11 at 3 53 15 PM"
src="https://github.com/getsentry/sentry/assets/83109586/384d76d8-5af4-45b9-b1b5-e1515c7c154b">

---------

Co-authored-by: Leander Rodrigues <leander.rodrigues@sentry.io>

cathteng committed:
feat(github-growth): link all repos task (#52595)

nikkikapadia committed:
fix(perf): add link to transaction summary on performance change explorer (#52730)

Addressing the feedback that @udameli and @narsaynorath gave about
having a link to the transaction summary from the performance change
explorer. This adds a link at the bottom of the performance change
explorer which directs users to the transaction summary with the trends
graph displayed.

<img width="500" alt="Screenshot 2023-07-12 at 12 11 35 PM"
src="https://github.com/getsentry/sentry/assets/72356613/cda93aea-e651-4b3c-96b0-8528b8757ebc">


EDIT: linked to transaction summary through open icon in header instead
<img width="500" alt="image"
src="https://github.com/getsentry/sentry/assets/72356613/6e7d9d39-6ed7-4768-af33-0e3fc1f9832f">

corps committed:
ref(hc): Support new notification settings serializer compatible with HC (#52834)

Extracts the backend half of
https://github.com/getsentry/sentry/pull/50279/files# to be deployed
separately from the frontend.

Duplicates the NotifcationSettings serializer and adds it as an optional
argument to endpoints that use it so that I can deploy the backend
change before the frontend change. Will remove and collapse once
frontend is out the door safely.

---------

Co-authored-by: Mark Story <mark@mark-story.com>

gggritso committed:
chore(starfish): Improve typing of `useSpansQuery` (#52904)

Related to [Abhi's recent
comment](https://github.com/getsentry/sentry/pull/52731#pullrequestreview-1526769790),
_one_ of the reasons we're causing runtime errors is that our types are
incorrect. This fixes _one_ spot, in which the response might be
undefined, and we don't type is correctly. There's a lot more digging to
do, but this already revealed some overly optimistic code.

markstory committed:
ref(hybridcloud) Remove validation on region addresses (#52827)

We're starting to spin up a pre-production environment for split silos
and want to use internal networking for region addresses so that we're
not going across the public WAN for RPC requests.

Removing this address validation allows us to do configure the
application this way. I've preserved the return value of Region.to_url()
using the public host names as this method is used to generate avatar
URLs.

azaslavsky committed:
feat(backup): Support comparators (#52597)

This change introduces the idea of comparators, or special handlers for
comparing JSON fields on exported models that cannot be tested for
simple stringh equivalency. The first of these comparators, the
`DateUpdatedComparator`, is added to ensure that comparisons on models
that have an audit timetamp incremented every time they are mutated are
still able to succeed.

rjo100 committed:
cleanup(crons): Removes crons issue platform flag (#52825)

Finishes cleanup from https://github.com/getsentry/sentry/pull/52036

malwilley committed:
feat(helpful-event): Better handling for bad contextual query (#52905)

Sometimes the helpful event endpoint can fail due to a bad query (or
potential bugs we have with querying), so this automatically removes the
query and retries without it. Since we probably want to investigate when
this occurs, so I'm logging a sentry error when it does.

Also disables automatic retries so we are able to quickly retry without
the query.

azaslavsky committed:
feat(backup): Support alert rules (#52669)

Adds a number of tests for exportable AlertRule* Sentry models, which
necessitates testing for SnubaQuery* and QuerySubscription models as
well, as these are included in AlertRule instances by default.

Issue: getsentry/team-ospo#156

leeandher committed:
ref(tests): Add webhook outbox testutil (#52821)

This PR adds a testutil for outbox testing that standardizes check for
outboxed webhook requests. It implements it across the existing parsers
as well.

hooray a  `+` < `-` diff

---------

Co-authored-by: Alberto Leal <mail4alberto@gmail.com>

azaslavsky committed:
test(backup): Ensure expected models are in output (#52690)

We add a new decorator, `@targets_models`, for the
`.../backup/test_models.py` test suite. The goal is two-fold: for each
individual test, the decorator provides a concise way to express which
models must be included in the output, lest we end up with a test that
passes the equality check, but only because it excluded our target
model(s) altogether. The second goal is to make the set of models being
exercised in the `ModelBackupTests` class easily visible to static
analysis tools like flake8, so that we may later create a check ensuring
that all `__include_in_export__ = True` marked models in this repository
are included in this test suite.

Issue: getsentry/team-ospo#156
Issue: getsentry/team-ospo#160

azaslavsky committed:
test(backup): Support dashboards (#52686)

Adds a number of tests for exportable Dashboard* Sentry models.

Commit:
feat(escalating-issues): Update Issue Stats UI (#52585)

## Objective:
Fixes https://github.com/getsentry/sentry/issues/51971

![sentry dev getsentry
net_7999_stats_issues_](https://github.com/getsentry/sentry/assets/10491193/fb9f25aa-1d1d-4956-a216-7feccca6ce81)

priscilawebdev committed:
ref(getting-started-docs): Migrate minidump doc to sentry main repo (#52871)

k-fish committed:
feat(perf): Add http/1.1 overhead detector (#52893)

### Summary
Adds a new detector (flags aren't going to be flipped until after
audit).

Servers using http/1.1 might be causing queueing behaviour on the
browser when the connections reach a certain amount. This detector
roughly detects overlapping spans against the same location (connections
are limited per host), assuming that queue depth is causing
monotonically increasing requestStart.

#### Other
- We'll likely need to change the network timing (`requestStart`) this
currently targets, but sdk needs to be cut before that.
- Our sdk collects absolute paths, so we skip url parsing in that case
and use location '/'
- Didn't want to have `/1.1` in filenames or enums so change the
file/vars to `http overhead`.
- We'll try a `500` delay threshold for now, meaning the peak of the
increasing times should be 500ms. I've found existing real events that
fire on this.

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

armenzg committed:
Address Anthony's feedback
